src/controller/CHIPDeviceController.cpp

```diff
@@ -1748,9 +1741,10 @@ void DeviceCommissioner::OnDeviceConnectedFn(void * context, Device * device)
     if (commissioner->mDeviceBeingPaired < kNumMaxActiveDevices)
     {
         Device * deviceBeingPaired = &commissioner->mActiveDevices[commissioner->mDeviceBeingPaired];
-        if (device == deviceBeingPaired && commissioner->mCommissioningStage == CommissioningStage::kFindOperational)
+        if (device == deviceBeingPaired && commissioner->mIsIPRendezvous)
```

Comment:
Shouldn't this check both conditions?  That is, the stage and the `mIsIPRendezvous` bit?

Reply:
No - that's the problem. CASE causes this to be called multiple times, including right after we send the commissioning complete command. That causes the OnCommissioningComplete function to be called - that was originally put there for BLE, which doesn't do a full commissioning. Calling that function causes the chip-tool to exit, without finishing off commissioning or cleaning up.

Reply:
Ah, so the point is that if `mIsIPRendezvous` we _do_ want to take the early return no matter what, so we don't reach the `OnCommissioningComplete` call?

We so need to fix #8010.....

Reply:
Yup - I'm actually working on that now - just finishing up the changes for chip-tool.